app/modules/intelligence/agents/custom_agents/custom_agent_model.py

```
@@ -25,5 +27,33 @@ class CustomAgent(Base):
     # Add foreign key constraint to users table
     __table_args__ = (ForeignKeyConstraint(["user_id"], ["users.uid"]),)
 
-    # Add relationship to User model
+    # Add relationships
     user = relationship("User", back_populates="custom_agents")
+    shares = relationship(
+        "CustomAgentShare", back_populates="agent", cascade="all, delete-orphan"
+    )
+
+    @property
+    def shared_with_users(self) -> List["User"]:
```

Ruff (0.8.2)

Undefined name `User`

(F821)